libs/labelbox/src/labelbox/schema/dataset.py

```
@@ -912,3 +913,80 @@ def _convert_items_to_upsert_format(self, _items):
+          >>>   dataset.set_iam_integration(iam_integration_id)
+          >>>
+          >>>    # Get IAM integration object
+          >>>    iam_integration = [integration.uid for integration
+          >>>      in iam_integrations
+          >>>      if integration.name == "My S3 integration"][0]
+          >>>
+          >>>   # Set IAM integration for IAMIntegrtion object
+          >>>   dataset.set_iam_integration(iam_integration)
+          >>>
+          >>>   # Unset IAM integration
+          >>>   dataset.set_iam_integration()
+          """
+
+          # Unset IAM integration if iam_integration is None
+          if iam_integration is None:
+              query = """mutation DetachSignerPyApi($id: ID!) {
+                  clearSignerForDataset(where: {id: $id}) {
+                      id
+                      signer {
+                      id
+                      }
+                  }
+                  }"""
+              response = self.client.execute(query, {"id": self.uid})
+              
+              if response:
+                 return response["clearSignerForDataset"]["signer"]
+              else:
+                raise LabelboxError("Can't unset IAM integration")
+              
+          else:
+
+              if isinstance(iam_integration, IAMIntegration):
+                  iam_integration_id = iam_integration.uid
+              else:
+                  iam_integration_id = iam_integration
+
+              query = """mutation AttachSignerPyApi($signerId: ID!, $datasetId: ID!) {
+                  setSignerForDataset(data: {signerId: $signerId}, where: {id: $datasetId}) {
+                      id
+                      signer {
+                      id
+                      }
+                  }
+                  }"""
+              response = self.client.execute(query, {"signerId": iam_integration_id, "datasetId": self.uid})
+              
+              # Return IAM Integration object if  
+              if response:
```

Response can only be `None` if the resource is not found. So I'd prefer something like:

```
if not response:
    <raise some appropriate ResourceNotFoundError>
    
<rest of code here>
```

fwiw I hate this pattern, `execute` should just throw an error instead of returning None (since that's what its docstring says it does) and I'd like to to refactor this as of seeing it in this PR. But for now, this pattern is present in the rest of client, so stick to this.

---

Done